src/bootstrap/builder.rs

```diff
@@ -746,6 +748,203 @@ impl<'a> Builder<'a> {
         StepDescription::run(v, self, paths);
     }
 
+    /// Modifies the interpreter section of 'fname' to fix the dynamic linker,
+    /// or the RPATH section, to fix the dynamic library search path
+    ///
+    /// This is only required on NixOS and uses the PatchELF utility to
+    /// change the interpreter/RPATH of ELF executables.
+    ///
+    /// Please see https://nixos.org/patchelf.html for more information
+    pub(crate) fn fix_bin_or_dylib(&self, fname: &Path) {
+        // FIXME: cache NixOS detection?
+        match Command::new("uname").arg("-s").stderr(Stdio::inherit()).output() {
+            Err(_) => return,
+            Ok(output) if !output.status.success() => return,
+            Ok(output) => {
+                let mut s = output.stdout;
+                if s.last() == Some(&b'\n') {
+                    s.pop();
+                }
+                if s != b"Linux" {
+                    return;
+                }
+            }
+        }
+
+        // If the user has asked binaries to be patched for Nix, then
+        // don't check for NixOS or `/lib`, just continue to the patching.
+        // FIXME: shouldn't this take precedence over the `uname` check above?
```

Comment:
The patching is not necessary on non-Linux systems. Patchelf doesn't work with Mach-O binaries like on macOS. On other unix systems there is a global system libc available that is not managed by Nix as other unix systems don't have a stable system call interface and thus only work with the system libc. Only on Linux can Nix manage libc and thus may have no global libc available.

Reply:
Ahh I see :) thanks for the clarification! I'll remove the FIXME.